engine/derivative.cc

```diff
@@ -647,6 +676,8 @@ namespace MathDAG
     throw error("vector derivatives not implemented");  \
   }                                                     
 
+  VECTOR_DERIVATIVE_NOT_IMPLEMENTED(gamma)
+  VECTOR_DERIVATIVE_NOT_IMPLEMENTED(digamma)
   VECTOR_DERIVATIVE_NOT_IMPLEMENTED(sum)
```

Comment:
These two shouldn't be here. But you will need to implement the recursive polygamma function.

Reply:
I would implement the derivative for gamma (which is very similar to the factorial one, of course), and then get rid of digamma as a  concept entirely (since polygamma with 0 as the second argument is equivalent).